File: backend/app/schemas/assistant.py
from __future__ import annotations

from pydantic import BaseModel, Field
from typing import List


class ChatRequest(BaseModel):
    message: str = Field(min_length=1, max_length=8000)
    top_k: int | None = None


class SourceSnippet(BaseModel):
    document_id: str
    filename: str
    version: int
    snippet: str


class ChatResponse(BaseModel):
    answer: str
    sources: List[SourceSnippet] = []
    mode: str

File: frontend/src/api/assistant.ts
import { api } from './client'

export type SourceSnippet = {
  document_id: string
  filename: string
  version: number
  snippet: string
}

export type ChatResponse = {
  answer: string
  sources: SourceSnippet[]
  mode: string
}

export async function chat(projectId: string, message: string, top_k?: number): Promise<ChatResponse> {
  const res = await api.post(`/projects/${projectId}/assistant/chat`, { message, top_k })
  return res.data
}
